stumpy/core.py

```diff
@@ -2494,3 +2494,40 @@ def _select_P_ABBA_value(P_ABBA, k, custom_func=None):
             MPdist = partition[k]
 
     return MPdist
+
+
+@njit(parallel=True)
+def _merge_topk_PI(PA, PB, IA, IB):
+    """
+    Merge two top-k matrix profiles PA and PB, and update PA (in place) while
+    prioritizing values of PA in ties. Also, update IA accordingly.
+
+    Parameters
+    ----------
+    PA : numpy.ndarray
+        a (top-k) matrix profile
+
+    PB : numpy.ndarray
+        a (top-k) matrix profile
+
+    IA : numpy.ndarray
+        a (top-k) matrix profile indices, corresponding to PA
+
+    IB : numpy.ndarray
+        a (top-k) matrix profile indices, corresponding to PB
+
+    Returns
+    -------
+    None
+    """
+    for i in prange(PA.shape[0]):
```

Comment:
I know that `PA` and `PB` should be the same size but I think we should change it to `for i in prange(PB.shape[0])` and `for j in range(PB.shape[1])`

Reply:
Sure. I will change both. May I ask why?

Reply:
Ahh...very nice! That makes sense!!! I will use PB.shape. Thanks for the explanation!